src/type/TypeInterface.php

```diff
@@ -59,4 +63,14 @@ public function equals(self $other): bool;
     public function matches(self $other): bool;
 
     public function isDefined(): bool;
+
+    public function isMonthly(): bool;
+
+    public function belongsToGroup(string $group): bool;
+
+    public function groupName(): string;
+
+    public function belongsToLocalCategory(string $local): bool;
+
+    public function localName(): string;
```

Comment:
This class is abstract and MUST NOT be aware of the actual type names, it can carry.

That's why we had `TypeSemantics` in a private repo, that solved this problem in a COMPOSITION pattern instead of the EXTENSION.